pkg/infra/iac2/templates/helm_chart/factory.ts

```diff
@@ -1,13 +1,47 @@
+import * as pulumi from '@pulumi/pulumi'
 import * as pulumi_k8s from '@pulumi/kubernetes'
 
 interface Args {
     Name: string
-    Directory: string
+    Directory?: pulumi.Input<string>
+    Chart?: pulumi.Input<string>
+    FetchOpts?: pulumi.Input<pulumi_k8s.helm.v3.FetchOpts>
+    Values?: pulumi.Inputs
+    Version?: pulumi.Input<string>
+    Namespace?: pulumi.Input<string>
+    KubernetesProvider: pulumi_k8s.Provider
+    dependsOn: pulumi.Input<pulumi.Input<pulumi.Resource>[]> | pulumi.Input<pulumi.Resource>
 }
 
 // noinspection JSUnusedLocalSymbols
 function create(args: Args): pulumi_k8s.helm.v3.Chart {
-    return new pulumi_k8s.helm.v3.Chart(args.Name, {
-        path: `./charts/${args.Directory}`,
-    })
+    return new pulumi_k8s.helm.v3.Chart(
+        args.Name,
+        {
+            //TMPL {{- if .Chart.Raw }}
+            chart: args.Chart,
+            //TMPL {{- end }}
+            //TMPL {{- if and (.Chart.Raw) (.FetchOpts.Raw) }}
+            fetchOpts: args.FetchOpts,
+            //TMPL {{- end }}
+            //TMPL {{- if not .Chart.Raw }}
+            path: `./charts/${args.Directory}`,
+            //TMPL {{- end }}
+            //TMPL {{- if .Namespace.Raw }}
+            namespace: args.Namespace,
+            //TMPL {{- end }}
+            //TMPL {{- if .Version.Raw }}
+            version: args.Version,
+            //TMPL {{- end }}
+            //TMPL {{- if .Values.Raw }}
+            values: args.Values,
+            //TMPL {{- end }}
+        },
```

Comment:
lol so ugly to look at, but also really nice that we can do this behind the scenes and it will look clean when its output

Reply:
Yeah. There's potential to build some higher order utility into the resource templates to automatically remove fields with a value of `null` or something like that.

Reply:
Could also do it as a function, something like:
```ts
//TMPL {{ argument "fetchOpts" .FetchOpts }}
```